geode-redis/src/main/java/org/apache/geode/redis/internal/data/RedisHash.java

```diff
@@ -45,10 +47,14 @@
 public class RedisHash extends AbstractRedisData {
   public static final RedisHash NULL_REDIS_HASH = new NullRedisHash();
   private HashMap<ByteArrayWrapper, ByteArrayWrapper> hash;
+  private HashMap<UUID, Map<ByteArrayWrapper,ByteArrayWrapper>> perClientHScanDataSnapShot;
```

Comment:
We should only be storing the list of keys here and not the whole hash. Redis doesn't guarantee the _value_ of a field remaining constant across scans just the list of fields. It's bad enough that we need to keep a list of fields but would be even worse if we also need to keep the values.

Reply:
ok-  changed to just store the keys